### ISSUE TITLE
Redirects do not need quoting

### ISSUE THREAD

**Issue by mina86** [koalaman/shellcheck#685]
Unnecessary warning when not quoting variables used in I/O redirects:

```
$ ls
a.sh
$ cat a.sh
name='a b'
echo >$name
$ sh a.sh
$ ls
a b  a.sh
$ shellcheck a.sh
In a.sh line 2:
echo >$name
      ^-- SC2086: Double quote to prevent globbing and word splitting.
$ shellcheck --version
ShellCheck - shell script analysis tool
version: 0.3.3
license: GNU Affero General Public License, version 3
website: http://www.shellcheck.net
```

**Comment by sanmai-NL** [comment]
What shell did you use?

With bash 4.3.42(1)

```bash
name='a b'
echo hi >$name
```

```
bash: $file2: ambiguous redirect
```

```bash
echo hi >"$name"
```

Using quotes is less surprising in general. 
> [Quoting is used to remove the special meaning of certain characters or words to the shell](http://pubs.opengroup.org/onlinepubs/009604499/utilities/xcu_chap02.html#tag_02_02). 

Suppose `name2='myfile'` and you have two files in your current directory, named literally `$name2` and `myfile`. `echo >$name2` will overwrite the latter, not the former. By single vs. double quoting you would optimally signal your intent to others less sure about what your script is trying to do. 

So it is better to change the warning to say that ‘quoting leads to less surprising results, e.g. due to globbing and word splitting’.

**Comment by mina86** [comment]
```
$ ls
a.sh
$ cat a.sh
name='a * b'
echo hi >$name
$ lksh -c 'echo $KSH_VERSION' && lksh a.sh && cat 'a * b' && rm 'a * b'
@(#)LEGACY KSH R46 2013/05/02
hi
$ mksh -c 'echo $KSH_VERSION' && mksh a.sh && cat 'a * b' && rm 'a * b'
@(#)MIRBSD KSH R46 2013/05/02
hi
$ ksh93 -c 'echo $KSH_VERSION' && ksh93 a.sh && cat 'a * b' && rm 'a * b'
Version AJM 93u+ 2012-08-01
hi
$ dpkg -s dash |grep Version && dash a.sh && cat 'a * b' && rm 'a * b'
Version: 0.5.7-4ubuntu1
hi
# dunno how to get version out of dash
$ busybox |head -n1 && busybox sh a.sh && cat 'a * b' && rm 'a * b'
BusyBox v1.21.1 (Ubuntu 1:1.21.0-1ubuntu1) multi-call binary.
hi
$ zsh5 --version && zsh5 a.sh && cat 'a * b' && rm 'a * b'
zsh 5.0.5 (x86_64-pc-linux-gnu)
hi
```
Having said that, I stand corrected since:
```
$ sh --version | head -n 1 && sh a.sh && cat 'a * b' && rm 'a * b'
GNU bash, version 4.3.11(1)-release (x86_64-pc-linux-gnu)
hi
$ bash --version | head -n 1 && bash --posix a.sh && cat 'a * b' && rm 'a * b'
GNU bash, version 4.3.11(1)-release (x86_64-pc-linux-gnu)
hi
```
but:
```
$ bash a.sh && cat 'a * b' && rm 'a * b'
a.sh: line 2: $name: ambiguous redirect
```
So it appears bash is broken. I’ve never noticed since I always write for `/bin/sh`.

Perhaps for this instance, the best wording would be ‘Double quote to prevent globbing and word splitting in bash’?

By the way, relevant part of the cited standard is (emphasis mine):
> For _[…]_ redirection operators _[other than here document]_, the word that follows the redirection operator shall be subjected to **tilde expansion, parameter expansion, command substitution, arithmetic expansion, and quote removal**. Pathname expansion shall not be performed on the word by a non-interactive shell; an interactive shell may perform it, but shall do so **only when the expansion would result in one word**.
